fbw-a32nx/src/wasm/systems/a320_systems/src/payload/mod.rs

```diff
@@ -0,0 +1,663 @@
+            pax_deboard_id,
+            pax_complete_id,
+            pax_ambience_id,
+            pax_boarding: false,
+            pax_deboarding: false,
+            pax_complete: false,
+            pax_ambience: false,
+        }
+    }
+
+    fn start_pax_boarding(&mut self) {
+        self.pax_boarding = true;
+    }
+
+    fn stop_pax_boarding(&mut self) {
+        self.pax_boarding = false;
+    }
+
+    fn start_pax_deboarding(&mut self) {
+        self.pax_deboarding = true;
+    }
+
+    fn stop_pax_deboarding(&mut self) {
+        self.pax_deboarding = false;
+    }
+
+    fn start_pax_complete(&mut self) {
+        self.pax_complete = true;
+    }
+
+    fn stop_pax_complete(&mut self) {
+        self.pax_complete = false;
+    }
+
+    fn start_pax_ambience(&mut self) {
+        self.pax_ambience = true;
+    }
+
+    fn stop_pax_ambience(&mut self) {
+        self.pax_ambience = false;
+    }
+}
+impl SimulationElement for A320BoardingSounds {
+    fn write(&self, writer: &mut SimulatorWriter) {
+        writer.write(&self.pax_board_id, self.pax_boarding);
+        writer.write(&self.pax_deboard_id, self.pax_deboarding);
+        writer.write(&self.pax_complete_id, self.pax_complete);
+        writer.write(&self.pax_ambience_id, self.pax_ambience);
+    }
+}
+
+pub struct A320Payload {
+    developer_state_id: VariableIdentifier,
+    is_boarding_id: VariableIdentifier,
+    board_rate_id: VariableIdentifier,
+    per_pax_weight_id: VariableIdentifier,
+
+    is_gsx_enabled_id: VariableIdentifier,
+    gsx_boarding_state_id: VariableIdentifier,
+    gsx_deboarding_state_id: VariableIdentifier,
+    gsx_pax_boarding_id: VariableIdentifier,
+    gsx_pax_deboarding_id: VariableIdentifier,
+    gsx_cargo_boarding_pct_id: VariableIdentifier,
+    gsx_cargo_deboarding_pct_id: VariableIdentifier,
+
+    developer_state: i8,
+    is_boarding: bool,
+    board_rate: BoardingRate,
+    per_pax_weight: Rc<Cell<Mass>>,
+
+    is_gsx_enabled: bool,
+    gsx_boarding_state: GsxState,
+    gsx_deboarding_state: GsxState,
+    gsx_pax_boarding: i32,
+    gsx_pax_deboarding: i32,
+    gsx_cargo_boarding_pct: f64,
+    gsx_cargo_deboarding_pct: f64,
+
+    pax: Vec<Pax>,
+    cargo: Vec<Cargo>,
+    boarding_sounds: A320BoardingSounds,
+    time: Duration,
+}
+impl A320Payload {
+    const DEFAULT_PER_PAX_WEIGHT_KG: f64 = 84.;
+    pub fn new(context: &mut InitContext) -> Self {
+        let per_pax_weight = Rc::new(Cell::new(Mass::new::<kilogram>(
+            Self::DEFAULT_PER_PAX_WEIGHT_KG,
+        )));
+
+        let mut pax = Vec::new();
+
+        for ps in A320Pax::iterator() {
+            pax.push(Pax::new(
+                context.get_identifier(A320_PAX[ps].pax_id.to_owned()),
+                context.get_identifier(format!("{}_DESIRED", A320_PAX[ps].pax_id).to_owned()),
+                context.get_identifier(A320_PAX[ps].payload_id.to_owned()),
+                Rc::clone(&per_pax_weight),
+            ));
+        }
```

Comment:
Linked to the comment above, you can directly use your enums here to construct the identifiers.

```suggestion
        for (id, ps) in A320Pax::iterator().enumerate() {
            pax.push(Pax::new(
                context.get_identifier(format!("{}", ps).to_owned()),
                context.get_identifier(format!("{}_DESIRED", ps).to_owned()),
                context.get_identifier(format!("PAYLOAD_STATION_{}_REQ", (id + 1)).to_owned()),
                Rc::clone(&per_pax_weight),
            ));
        }
```

Reply:
Note for `cargo` you'll need to have `id + 5` instead of 1.

Reply:
I'll try and think of a better solution for this overall, perhaps in this PR or in the next refactor